src/commands/react-native/upload.ts

```diff
@@ -0,0 +1,319 @@
+
+      return 1
+    }
+
+    this.context.stdout.write(
+      renderCommandInfo(
+        this.bundle,
+        this.sourcemap,
+        this.platform,
+        this.releaseVersion,
+        this.service,
+        this.maxConcurrency,
+        this.dryRun,
+        this.projectPath,
+        this.buildVersion
+      )
+    )
+    const metricsLogger = getMetricsLogger({
+      datadogSite: process.env.DATADOG_SITE,
+      defaultTags: [
+        `version:${this.releaseVersion}`,
+        `build:${this.buildVersion}`,
+        `service:${this.service}`,
+        `cli_version:${this.cliVersion}`,
+        'react-native:true',
+        `platform:${this.platform}`,
+      ],
+      prefix: 'datadog.ci.sourcemaps.upload.',
+    })
+    const apiKeyValidator = newApiKeyValidator({
+      apiKey: this.config.apiKey,
+      datadogSite: this.config.datadogSite,
+      metricsLogger: metricsLogger.logger,
+    })
+    const useGit = this.disableGit === undefined || !this.disableGit
+    const initialTime = Date.now()
+    const payloads = await this.getPayloadsToUpload(useGit)
+    const requestBuilder = this.getRequestBuilder()
+    const uploadMultipart = this.upload(requestBuilder, metricsLogger, apiKeyValidator)
+    try {
+      const results = await asyncPool(this.maxConcurrency, payloads, uploadMultipart)
+      const totalTime = (Date.now() - initialTime) / 1000
+      this.context.stdout.write(renderSuccessfulCommand(results, totalTime, this.dryRun))
+      metricsLogger.logger.gauge('duration', totalTime)
+
+      return 0
+    } catch (error) {
+      if (error instanceof InvalidConfigurationError) {
+        this.context.stdout.write(renderConfigurationError(error))
+
+        return 1
+      }
+      // Otherwise unknown error, let's propagate the exception
+      throw error
+    } finally {
+      try {
+        await metricsLogger.flush()
+      } catch (err) {
+        this.context.stdout.write(`WARN: ${err}\n`)
+      }
+    }
+  }
+
+  // Fills the 'repository' field of each payload with data gathered using git.
+  private addRepositoryDataToPayloads = async (payloads: RNSourcemap[]) => {
+    const repositoryData = await getRepositoryData(await newSimpleGit(), this.context.stdout, this.repositoryURL)
+    if (repositoryData === undefined) {
+      return
+    }
+    await Promise.all(
+      payloads.map(async (payload) => {
+        const repositoryPayload = this.getRepositoryPayload(repositoryData!, payload.sourcemapPath)
+        payload.addRepositoryData({
+          gitCommitSha: repositoryData.hash,
+          gitRepositoryPayload: repositoryPayload,
+          gitRepositoryURL: repositoryData.remote,
+        })
+      })
+    )
+  }
+
+  // Looks for the sourcemaps and minified files on disk and returns
+  // the associated payloads.
+  private getMatchingRNSourcemapFiles = async (): Promise<RNSourcemap[]> => [
+    new RNSourcemap(this.bundle!, this.sourcemap!),
+  ]
```

Comment:
This function and the one below should return the type `RNSourcemap`, not `Promise<RNSourcemap[]>`

Reply:
For the one below, it is actually async because we execute a git command in `this.addRepositoryDataToPayloads` -> `getRepositoryData` -> git commands